client/src/js/errors/reducer.js

```diff
@@ -0,0 +1,87 @@
+import { endsWith, replace } from "lodash-es";
+import { reportAPIError } from "../utils";
+import {
+    CLEAR_ERROR,
+    CREATE_SAMPLE,
+    UPDATE_SAMPLE,
+    CREATE_VIRUS,
+    EDIT_VIRUS,
+    ADD_ISOLATE,
+    EDIT_ISOLATE,
+    ADD_SEQUENCE,
+    EDIT_SEQUENCE,
+    CREATE_INDEX,
+    CREATE_SUBTRACTION,
+    UPDATE_ACCOUNT,
+    CHANGE_ACCOUNT_PASSWORD,
+    CREATE_USER,
+    EDIT_USER,
+    CREATE_GROUP
+} from "../actionTypes";
+
+const checkActionFailed = (action) => {
+    const isFailed = endsWith(action.type, "_FAILED");
+    return isFailed ? action : false;
+};
+
+const getErrorName = (action) => {
+    const errorName = replace(action.type, "_FAILED", "_ERROR");
+    return errorName;
+};
+
+const resetErrorName = (newAction) => {
+    const errorName = endsWith(newAction.type, "_REQUESTED")
+        ? replace(newAction.type, "_REQUESTED", "_ERROR")
+        : null;
+
+    return errorName;
+};
+
+export default function errorsReducer (state = null, action) {
+
+    const failedAction = checkActionFailed(action);
+    let errorName;
+
+    if (failedAction) {
+        errorName = getErrorName(action);
+    } else if (action.type === CLEAR_ERROR) {
```

Comment:
The `if (action.type === CLEAR_ERROR)` conditional should be moved to the top of the reducer.

This is a quick check that will let you avoid calling `checkActionFailed()` unnecessarily.